custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/scala/com/datarobot/custom/H2OPredictor.scala

```diff
@@ -0,0 +1,199 @@
+        case true  => Array(this.model.predictRegression(record).value)
+        case false => this.model.predictBinomial(record).classProbabilities
+      }
+      prediction
+    }.toArray
+    predictions
+  }
+
+  def map2RowData(x: java.util.Map[String, String]): RowData = {
+    val row = x.asScala
+    val rowData: RowData = new RowData();
+    row.foreach { case (k, v) => rowData.put(k, v.toString) }
+    rowData
+  }
+
+  def modelConfigViaMojo(mojo: java.io.File): EasyPredictModelWrapper.Config = {
+    new EasyPredictModelWrapper.Config()
+      .setModel(MojoModel.load(mojo.getAbsolutePath))
+      .setConvertUnknownCategoricalLevelsToNa(true)
+      .setConvertInvalidNumbersToNa(true)
+  }
+
+  // this would work with a properly compiled / packaged pojo
+  // this will work with a compile pojo
+  // the expectation is that we compile the pojo when the container is spun up.
+// s"javac -cp code/h2o-genmodel-3.30.0.6.jar ${pojo.getAbsolutePath}".!
+
+  def modelConfigViaPojo(
+      pojo: java.io.File
+  ): EasyPredictModelWrapper.Config = {
+    val urls = pojo.getName.endsWith("java") match {
+      case true => {
+        val pathOfJar = classOf[hex.genmodel.GenModel].getProtectionDomain.getCodeSource.getLocation.toURI.getPath
+        s"javac -cp ${pathOfJar} ${pojo.getAbsolutePath}".!
+        Array(pojo.getAbsoluteFile.getParentFile.toURI.toURL)
+      }
+      case false => throw new Exception("Pojo not availabe")
+    }
+
+    val pojoName = pojo.getName.replace(".jar", "").replace(".java", "")
+    val urlClassLoader = URLClassLoader.newInstance(
+      urls,
+      Thread.currentThread().getContextClassLoader()
+    )
+    val h2oRawPredictor =
+      urlClassLoader.loadClass(pojoName).newInstance.asInstanceOf[GenModel]
+    val modelConfig = new EasyPredictModelWrapper.Config()
+      .setModel(h2oRawPredictor)
+      .setConvertUnknownCategoricalLevelsToNa(true)
+      .setConvertInvalidNumbersToNa(true)
+    // EasyPredictModelWrapper = new EasyPredictModelWrapper(modelConfig)
+    modelConfig
+  }
+
+  def loadModel(modelDir: String): EasyPredictModelWrapper = {
+
+    // h2o model artifacts
+    val re = new scala.util.matching.Regex("(.java$)|(.zip$)")
+
+    val files = Paths.get(modelDir).toFile().listFiles().filter { f =>
+      re.findAllMatchIn(f.getName).hasNext
+    }
+
+    val file = files.length match {
+      case 0 =>
+        throw new Exception("no model artifact found in model directory")
+      case 1 => files.apply(0)
+      case _ =>
+        throw new Exception(
+          s"more than 1 main model artifact found in model directory: ${files.map { _.getName }.mkString(",")}"
+        )
+    }
+
+    val fileExt = re.findAllIn(file.getName).next
+
+    val modelConfig = fileExt match {
+      case ".zip" => modelConfigViaMojo(file)
+      case ".java"  => modelConfigViaPojo(file)
+      case _ => throw new Exception("""|no usable H2O model artifact available""".stripMargin)
```

Comment:
what does the weird quotes and vertical line and stripmargin do.

Reply:
nothing special - useful for multiline strings in scala.  it is superfluous here

Reply:
removed this formatting